lib/primitives/base-element.js

```diff
@@ -132,6 +135,19 @@ module.exports = function(registry) {
 
     setMetaProperty: function(name, value) {
       this.meta.set(name, value);
+    },
+
+    emitSwap: function() {
```

Comment:
This does not seem to correlate the objects related to a particular event as to what was replaced by what if I understand this correctly. Seems like something is missing with respect to the event. Should this have some event object that contains the information?